packages/govuk-frontend/src/govuk/components/header/header.mjs

```diff
@@ -52,22 +52,43 @@ export class Header extends GOVUKFrontendComponent {
     }
 
     this.$module = $module
-    this.$menuButton = $module.querySelector('.govuk-js-header-toggle')
-    this.$menu =
-      this.$menuButton &&
-      $module.querySelector(
-        `#${this.$menuButton.getAttribute('aria-controls')}`
-      )
-
-    if (
-      !(
-        this.$menuButton instanceof HTMLElement ||
-        this.$menu instanceof HTMLElement
-      )
-    ) {
+    const $menuButton = $module.querySelector('.govuk-js-header-toggle')
+
+    // Headers don't necessarily have a navigation.
+    // When they don't, the menu toggle won't be rendered
+    // by our macro (or may be omitted when writing plain HTML)
+    if (!$menuButton) {
       return this
     }
 
+    const menuId = $menuButton.getAttribute('aria-controls')
+    if (!menuId) {
+      throw new ElementError($menuButton, {
+        componentName: 'Header',
+        identifier: '$menuButton["aria-controls"]',
+        expectedType: 'string'
+      })
+    }
+
+    const $menu = document.getElementById(menuId)
+
+    if (!($menuButton instanceof HTMLElement)) {
+      throw new ElementError($menu, {
+        componentName: 'Header',
+        identifier: 'Menu'
+      })
```

Comment:
Sorry for not spotting this earlier

We'll need to customise this one for the button not the menu:

```suggestion
      throw new ElementError($menuButton, {
        componentName: 'Header',
        identifier: '.govuk-js-header-toggle'
      })
```

Reply:
Whoops, forgot to update when I copy-pasted, sorry.

Reply:
Don't worry, do you want to add tests to make sure we've caught them all?